web/src/tenor.ts

```diff
@@ -104,8 +113,9 @@ function render_featured_gifs(next_page: boolean): void {
     if (is_loading_more || (current_search_term !== undefined && current_search_term.length > 0)) {
         return;
     }
-    let data = {
+    let data: Record<string, undefined | number | string> = {
         ...base_payload,
+        contentfilter: tenor_rating_map[get_rating()],
```

Comment:
Shouldn't this just go into the `base_payload` construction?

Oh, you have `base_payload` not a function. I think that's a bug; various properties in it, like the default_language, are mutable. So I think you can just make it a `get_base_payload` function and then you can just add this line there rather than duplicating it in both places.